### PR TITLE
chore: Update references to ConnectorPluginInfo to PluginInfo

### PR DESCRIPTION
**Pull request by jnh5y**
Addresses https://github.com/confluentinc/ksql/issues/8841 / https://github.com/confluentinc/ksql/issues/8863

### Description 
This follows up on a change in Apache Kafka here: https://github.com/apache/kafka/commit/029a14b530dcaab980faee4fa18c925b95db3122

### Testing done 
Unit test updates should be sufficient.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by jnh5y** [comment]
CI will fail until https://github.com/apache/kafka/commit/029a14b530dcaab980faee4fa18c925b95db3122 lands in CCS.

I also ran through the serialization work that was required to get the `DefaultConnectClientTest` to pass, so there may be a better way to do that.

File: pom.xml [hunk 1]
```diff
@@ -155,6 +155,7 @@
         <netty.version>4.1.72.Final</netty.version>
         <netty-codec-http2-version>4.1.72.Final</netty-codec-http2-version>
         <jersey-common>2.34</jersey-common>
+        <kafka.version>3.2.0-SNAPSHOT</kafka.version>
```
Comment (lct45) [inline pom.xml:158]: nit: should this be here?

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java [hunk 2]
```diff
@@ -43,18 +44,21 @@ public StatementExecutorResponse execute(
...
         serviceContext.getConnectClient().connectorPlugins();
     if (plugins.error().isPresent()) {
       return StatementExecutorResponse.handled(connectErrorHandler.handle(
           configuredStatement, plugins));
     }
 
     final List<SimpleConnectorPluginInfo> pluginInfos = new ArrayList<>();
-    for (final ConnectorPluginInfo info : plugins.datum().get()) {
+    for (final PluginInfo info : plugins.datum().get()) {
+      System.out.println("Info.type(): " + info.type() + " ConnectorType: "
```
Comment (lct45) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java:56]: Is this meant to be here?

**Review by lct45 — APPROVED** [review]
two nits, otherwise lgtm thanks @jnh5y

File: pom.xml (hunk 1 above)
Comment (jnh5y) [reply pom.xml:158]: Nope!  Good catch.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java (hunk 2 above)
Comment (jnh5y) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java]: Nope.  Was rushing yesterday to have something ready.:).

**Pull request merged**